### PR TITLE
XFA - Support aria heading and table structure. (bug 1723421) (bug 1723425)

### PR DESCRIPTION
**Pull request by brendandahl**
https://bugzilla.mozilla.org/show_bug.cgi?id=1723421
https://bugzilla.mozilla.org/show_bug.cgi?id=1723425

### CODE REVIEW

**Review by marco-c — APPROVED** [review]

File: src/core/xfa/template.js [hunk 1]
```diff
@@ -1833,9 +1835,20 @@ class Draw extends XFAObject {
       children: [],
     };
 
-    const assist = this.assist ? this.assist[$toHTML]() : null;
+    const assist = this.assist;
```
Comment (calixteman) [inline src/core/xfa/template.js:1838]: You probably want to add the same code for `exclGroup`, `field` and `subform`.

File: src/core/xfa/template.js [hunk 2]
```diff
@@ -1833,9 +1835,20 @@ class Draw extends XFAObject {
...
     if (assist) {
-      html.attributes.title = assist;
+      const assistHTML = this.assist[$toHTML]();
```
Comment (Snuffleupagus) [inline src/core/xfa/template.js:1840]: I might be misunderstanding the code, but would the slightly more compact `const assistHTML = assist[$toHTML]();` work?

File: test/unit/xfa_tohtml_spec.js [hunk 3]
```diff
@@ -177,6 +177,44 @@ describe("XFAFactory", function () {
...
+      `;
+      const factory = new XFAFactory({ "xdp:xdp": xml });
+
+      expect(factory.numberPages).toEqual(1);
+
+      const pages = factory.getPages();
+      const page1 = pages.children[0];
+      const wrapper = page1.children[0];
+      const draw = wrapper.children[0];
+      console.log(draw);
```
Comment (Snuffleupagus) [inline test/unit/xfa_tohtml_spec.js:212]: This looks like a left-over debug statement, please remove it.

**Review by calixteman — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**